orchestrator/graphql/resolvers/process.py

```diff
@@ -0,0 +1,224 @@
+                        .filter(ProcessSubscriptionTable.workflow_target.in_(targets))
+                        .subquery()
+                    )
+                    query = query.filter(ProcessTable.pid == process_subscriptions.c.pid)
+                else:
+                    raise_status(HTTPStatus.BAD_REQUEST, f"Invalid filter '{field}'")
+    return query
+
+
+def sort_processes(query: SearchQuery, sort_by: list[ProcessSort] | None = None) -> SearchQuery:
+    if sort_by is not None:
+        for item in sort_by:
+            if item.field in VALID_SORT_KEYS:
+                sort_key = VALID_SORT_KEYS[item.field]
+                if item.order == SortOrder.DESC:
+                    query = query.order_by(expression.desc(ProcessTable.__dict__[sort_key]))
+                else:
+                    query = query.order_by(expression.asc(ProcessTable.__dict__[sort_key]))
+            else:
+                raise_status(HTTPStatus.BAD_REQUEST, "Invalid Sort parameters")
+    return query
+
+
+def set_processes_range(query: SearchQuery, after: int, first: int) -> SearchQuery:
+    if after is not None and first:
+        if after >= after + first:
+            msg = "range start must be lower than end"
+            logger.exception(msg)
+            raise_status(HTTPStatus.BAD_REQUEST, msg)
+        query = query.offset(after).limit(first + 1)
+    return query
+
+
+async def resolve_processes(
+    filter_by: list[tuple[str, str]] | None = None,
+    sort_by: list[ProcessSort] | None = None,
+    first: int = 10,
+    after: int = 0,
+) -> Connection[ProcessType]:
+    _range: list[int] | None = [after, after + first] if after is not None and first else None
+    _filter: list[str] | None = list(flatten(filter_by)) if filter_by else None
+    logger.info("processes_filterable() called", range=_range, sort=sort_by, filter=_filter)
+
+    # the joinedload on ProcessSubscriptionTable.subscription via ProcessBaseSchema.process_subscriptions prevents a query for every subscription later.
+    # tracebacks are not presented in the list of processes and can be really large.
+    query = ProcessTable.query.options(
+        joinedload(ProcessTable.process_subscriptions)
+        .joinedload(ProcessSubscriptionTable.subscription)
+        .joinedload(SubscriptionTable.product),
+        defer("traceback"),
+    )
+
+    query = filter_processes(query, _filter)
+    query = sort_processes(query, sort_by)
+    total = query.count()
+    query = set_processes_range(query, after, first)
+
+    processes = query.all()
+    has_next_page = len(processes) > first
+
+    # exclude last item as it was fetched to know if there is a next page
+    processes = processes[:first]
+    processes_length = len(processes)
+    start_cursor = after if processes_length else None
+    end_cursor = after + processes_length - 1
+
+    return Connection(
+        page=[ProcessType.from_pydantic(enrich_process(p)) for p in processes] if processes else [],
```

Comment:
Maybe assign to variables so you  can just have 'return Connection(page=page, page_info=page_info)' here